pkg/cmd/init.go

```diff
@@ -0,0 +1,338 @@
+				port = "443"
+				insecure-flag = "0"
+
+				[VirtualCenter "<VCENTER-ADDRESS>"]
+
+				[Workspace]
+				server = "<VCENTER-ADDRESS>"
+				datacenter = "<DATACENTER>"
+				default-datastore="<DATASTORE>"
+				resourcepool-path=""
+				folder = ""
+
+				[Disk]
+				scsicontrollertype = pvscsi
+
+				[Network]
+				public-network = "<VM-NETWORK>"
+			`),
+			csiConfig: heredoc.Doc(`
+				[Global]
+				cluster-id = "<CLUSTER-ID>"
+				cluster-distribution = "<CLUSTER-DISTRIBUTION>"
+
+				[VirtualCenter "<VCENTER-ADDRESS>"]
+				insecure-flag = "false"
+				user = "<USERNAME>"
+				password = "<PASSWORD>"
+				port = "<PORT>"
+				datacenters = "<DATACENTER>"
+			`),
+		},
+	}
+)
+
+type initOpts struct {
+	Provider  oneOfFlag `longflag:"provider"`
+	Name      string    `longflag:"name"`
+	Version   string    `longflag:"version"`
+	Terraform bool      `longflag:"terraform"`
+	Path      string    `longflag:"path"`
+}
+
+func initCmd() *cobra.Command {
+	opts := &initOpts{
+		Provider: oneOfFlag{
+			validSet:     sets.StringKeySet(validProviders),
+			defaultValue: "none",
+		},
+	}
+
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "init new kubeone cluster configuration",
+		Long: heredoc.Doc(`
+			Initialize new KubeOne + terraform configuration for chosen provider.
+		`),
+		SilenceErrors: true,
+		Example:       `kubeone init --provider aws`,
+		RunE: func(_ *cobra.Command, args []string) error {
+			return runInit(opts)
+		},
+	}
+
+	providerUsageText := fmt.Sprintf("provider to initialize, possible values: %s", strings.Join(opts.Provider.PossibleValues(), ", "))
+
+	cmd.Flags().BoolVar(&opts.Terraform, longFlagName(opts, "Terraform"), false, "generate terraform config")
+	cmd.Flags().StringVar(&opts.Name, longFlagName(opts, "Name"), "example", "name of the cluster")
+	cmd.Flags().StringVar(&opts.Version, longFlagName(opts, "version"), "v1.24.5", "kubernetes version")
+	cmd.Flags().StringVar(&opts.Path, longFlagName(opts, "Path"), ".", "path where to write files")
+	cmd.Flags().Var(&opts.Provider, longFlagName(opts, "Provider"), providerUsageText)
+
+	return cmd
+}
+
+func runInit(opts *initOpts) error {
+	err := os.MkdirAll(opts.Path, 0750)
+	if err != nil {
+		return err
+	}
+
+	k1config, err := os.OpenFile(filepath.Join(opts.Path, "kubeone.yaml"), os.O_CREATE|os.O_WRONLY, 0600)
+	if err != nil {
+		return fail.Runtime(err, "creating manifest file")
+	}
+	defer k1config.Close()
+
+	ybuf, err := genKubeOneClusterYAML(opts)
+	if err != nil {
+		return fail.Runtime(err, "generating KubeOneCluster")
+	}
+
+	_, err = io.Copy(k1config, bytes.NewBuffer(ybuf))
+	if err != nil {
+		return fail.Runtime(err, "writing KubeOneCluster")
+	}
+
+	if opts.Terraform {
+		prov := validProviders[opts.Provider.String()]
+		if err = examples.CopyTo(opts.Path, prov.terraformPath); err != nil {
+			return fail.Runtime(err, "copying terraform configuration")
+		}
+	}
+
+	return nil
+}
+
+func genKubeOneClusterYAML(opts *initOpts) ([]byte, error) {
```

Comment:
meh, I'm absolutely not fan of it

Reply:
Neither I am, to be honest. But I think it's more important what we deliver to users. And since the manifest is quite simple, I don't think it's a huge issue to use templating instead. We can discuss that later on.